app/src/main/java/com/example/sharingang/NewEditFragment.kt

```diff
@@ -161,10 +174,45 @@ class NewItemFragment : Fragment() {
     }
 
     private fun updateLocation(location: Location) {
-        binding.latitude = location.latitude.toString()
-        binding.longitude = location.longitude.toString()
+        updateLocationWithCoordinates(location.latitude, location.longitude)
+    }
+
+    private fun updateLocationWithCoordinates(latitude: Double, longitude: Double) {
+        binding.latitude = latitude.toString()
+        binding.longitude = longitude.toString()
         val address =
-            geocoder.getFromLocation(location.latitude, location.longitude, 1).getOrNull(0)
+            geocoder.getFromLocation(latitude, longitude, 1).getOrNull(0)
         binding.postalAddress.text = address?.getAddressLine(0) ?: ""
     }
+
+    private fun setupNewOrEditFragment() {
+        if (viewModel.onEdit.value == false) {
+            binding.editItemPrompt.visibility = View.GONE
+            binding.editItemButton.visibility = View.GONE
+            binding.newItemPrompt.visibility = View.VISIBLE
+            binding.createItemButton.visibility = View.VISIBLE
+        } else {
+            binding.editItemPrompt.visibility = View.VISIBLE
+            binding.editItemButton.visibility = View.VISIBLE
+            binding.newItemPrompt.visibility = View.GONE
+            binding.createItemButton.visibility = View.GONE
```

Comment:
Maybe it would be easier to work with only one prompt and one button and change their appearance/behavior depending on whether an item is passed as an argument. Instead of having two of each, but I don't know if it's easier and clearer.

Reply:
By doing so we would need to change their appearance and their action. I thought it was easier this way. Also, if we want to change the GUI later one, we can do it directly in the XML file, without going through the fragment.